src/extension/sdk/utils.ts

```diff
@@ -539,6 +547,50 @@ export class SdkUtils {
 		);
 	}
 
+	private async runCustomGetSDKCommand(command: GetSDKCommandConfig): Promise<string | undefined> {
+		const globalWorkDir = this.getGlobalWorkingDirectory();
+		if (!globalWorkDir) return undefined;
+
+		const cmdWorkDir = command.cwd ?? ".";
+		const cmdWorkDirAbs = path.isAbsolute(cmdWorkDir) ? cmdWorkDir : path.join(globalWorkDir, cmdWorkDir);
+		try {
+			const commandResult = await runToolProcess(this.logger, cmdWorkDirAbs, command.executable, command.args ?? [], command.env);
+			if (commandResult.exitCode !== 0) {
+				throw new Error("Command exited with non-zero code");
+			}
+			const sdkPath = commandResult.stdout.trim();
+			if (!sdkPath)
+				throw new Error(`Command output was empty`);
+			return sdkPath;
+		}catch (e) {
+			const commandJson = JSON.stringify(command);
+			this.logger.error(`Failed to run the command to get the SDK: ${e}. Command: ${commandJson}. CWD: ${cmdWorkDirAbs}`);
+			return undefined;
+		}
+	}
+
+	/**
+	 * Obtains a sane default as a working directory.
+	 * When using a multiroot workspace setup, return the directory of the workspace file
+	 * Otherwise, return the path of the workspace directory.
+	 * If there is no workspace directory, return undefined
+	 */
+	private getGlobalWorkingDirectory(): string | undefined {
```

Comment:
> Working directory on the command:
> Because the SDK is obtained in global context of the extension, we need to infer where we want to run the command. I created a separate function that sets a different work dir based on the vscode workspace config (single/multi root). Are you ok with the current implementation?

I think this is reasonable - although the word "global" here confuses me, maybe something like `getWorkingDirectoryForGetSdkCommand` would be better?